hazelcast/src/test/java/com/hazelcast/client/usercodedeployment/ClientUserCodeDeploymentTest.java

```diff
@@ -158,6 +161,46 @@ public void testWithMultipleNodes_clientReconnectsToNewNode() throws Interrupted
         assertCodeDeploymentWorking(client, new IncrementingEntryProcessor());
     }
 
+    @Test
+    public void testWithSplitBrain_clientReconnectsToOtherHalf() {
```

Comment:
More explicit name like `testDeploysWhenReconnectsToOtherHalfDuringSplitBrain` could be better.